core/src/main/java/org/elasticsearch/common/xcontent/support/filtering/FilterPathBasedFilter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent.support.filtering;
+
+import com.fasterxml.jackson.core.filter.TokenFilter;
+import org.elasticsearch.common.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FilterPathBasedFilter extends TokenFilter {
+
+    /**
+     * Marker value that should be used to indicate that a property name
+     * or value matches one of the filter paths.
+     */
+    private final static TokenFilter MATCHING = new TokenFilter(){};
+
+    /**
+     * Marker value that should be used to indicate that none of the
+     * property names/values matches one of the filter paths.
+     */
+    private final static TokenFilter NO_MATCHING = new TokenFilter(){};
+
+    private final FilterPath[] filters;
+
+    public FilterPathBasedFilter(FilterPath[] filters) {
+        if (CollectionUtils.isEmpty(filters)) {
+            throw new IllegalArgumentException("filters cannot be null or empty");
+        }
+        this.filters = filters;
+    }
+
+    public FilterPathBasedFilter(String[] filters) {
+        this(FilterPath.compile(filters));
+    }
+
+    /**
+     * Evaluates if a property name matches one of the given filter paths.
+     */
+    private TokenFilter evaluate(String name, FilterPath[] filters) {
+        if (filters != null) {
```

Comment:
In case you don't need the parameter, you can remove the null check as the constructor already checks that as a precondition.